extra/modules/greenbids-real-time-data/src/main/java/org/prebid/server/hooks/modules/greenbids/real/time/data/core/GreenbidsInvocationService.java

```diff
@@ -61,10 +64,15 @@ private Boolean isExploration(GreenbidsConfig greenbidsConfig, String greenbidsI
 
     private List<Imp> updateImps(BidRequest bidRequest, Map<String, Map<String, Boolean>> impsBiddersFilterMap) {
         return bidRequest.getImp().stream()
+                .filter(imp -> isImpKept(imp, impsBiddersFilterMap))
                 .map(imp -> updateImp(imp, impsBiddersFilterMap.get(imp.getId())))
                 .toList();
     }
 
+    private boolean isImpKept(Imp imp, Map<String, Map<String, Boolean>> impsBiddersFilterMap) {
+        return impsBiddersFilterMap.get(imp.getId()).values().stream().anyMatch(isKept -> isKept);
```

Comment:
It seems that it is time for us to change the structure from `Map<String, Map<String, Boolean>>` to `Map<String, Set<String>> impIdToKeptBidders` (from the moment of creation in the `processProbabilities` method).

This is only a suggestion, it is not necessary to change it now.